crates/witgen_macro_helper/src/lib.rs

```diff
@@ -0,0 +1,241 @@
+#![deny(warnings)]
+use std::{
+    fs::{self, OpenOptions},
+    io::Write,
+    path::{Path, PathBuf},
+};
+
+use anyhow::{bail, Context, Result};
+use heck::ToKebabCase;
+
+use quote::ToTokens;
+use syn::Type;
+
+mod generator;
+pub use generator::{
+    gen_wit_enum, gen_wit_function, gen_wit_struct, gen_wit_type_alias, get_target_dir,
+};
+
+use once_cell::sync::OnceCell;
+
+static TARGET_PATH: OnceCell<PathBuf> = OnceCell::new();
+
+#[macro_export]
+macro_rules! handle_error {
+    ($op: expr) => {
+        let content = match $op {
+            Ok(res) => res,
+            Err(err) => {
+                return syn::Error::new(Span::call_site(), format!("witgen error: {}", err))
+                    .to_compile_error()
+                    .into();
+            }
+        };
+        let target_dir = witgen_macro_helper::set_target_dir();
+
+        if let Err(err) = witgen_macro_helper::write_to_file(&target_dir, content) {
+            return syn::Error::new(
+                Span::call_site(),
+                format!("witgen error: cannot write to file ({})", err),
+            )
+            .to_compile_error()
+            .into();
+        };
+    };
+}
+
+pub fn set_target_dir() -> PathBuf {
```

Comment:
If we decide to keep all these functions related to file generation in this crate then apply this change please, it will be more convenient.
```suggestion
#[doc(hidden)]
pub fn get_or_init_target_dir() -> PathBuf {
```